provider/plugin/plugin.go

```diff
@@ -0,0 +1,350 @@
+		return true
+	}
+
+	req, err := http.NewRequest("POST", u, bytes.NewBuffer(b))
+	if err != nil {
+		propertyValuesEqualErrorsGauge.Inc()
+		log.Debugf("Failed to create request: %s", err)
+		return true
+	}
+	req.Header.Set(acceptHeader, mediaTypeFormatAndVersion)
+	resp, err := p.client.Do(req)
+	if err != nil {
+		propertyValuesEqualErrorsGauge.Inc()
+		log.Debugf("Failed to perform request: %s", err)
+		return true
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode != http.StatusOK {
+		propertyValuesEqualErrorsGauge.Inc()
+		log.Debugf("Failed to run PropertyValuesEqual with code %d", resp.StatusCode)
+		return true
+	}
+
+	respoBody, err := io.ReadAll(resp.Body)
+	if err != nil {
+		propertyValuesEqualErrorsGauge.Inc()
+		log.Errorf("Failed to read body: %v", err)
+		return true
+	}
+	r := PropertyValuesEqualResponse{}
+	err = json.Unmarshal(respoBody, &r)
+	if err != nil {
+		propertyValuesEqualErrorsGauge.Inc()
+		log.Errorf("Failed to unmarshal body: %v", err)
+		return true
+	}
+	return r.Equals
+}
+
+// AdjustEndpoints will call the provider doing a POST on `/adjustendpoints` which will return a list of modified endpoints
+// based on a provider specific requirement.
+// This method returns an empty slice in case there is a technical error on the provider's side so that no endpoints will be considered.
+func (p PluginProvider) AdjustEndpoints(e []*endpoint.Endpoint) []*endpoint.Endpoint {
+	endpoints := []*endpoint.Endpoint{}
+	u, err := url.JoinPath(p.remoteServerURL.String(), "adjustendpoints")
+	if err != nil {
+		adjustEndpointsErrorsGauge.Inc()
+		log.Debugf("Failed to join path, %s", err)
+		return endpoints
+	}
+	b, err := json.Marshal(e)
+	if err != nil {
+		adjustEndpointsErrorsGauge.Inc()
+		log.Debugf("Failed to marshal endpoints, %s", err)
+		return endpoints
+	}
+	req, err := http.NewRequest("POST", u, bytes.NewBuffer(b))
+	if err != nil {
+		adjustEndpointsErrorsGauge.Inc()
+		log.Debugf("Failed to create new HTTP request, %s", err)
+		return endpoints
+	}
+	req.Header.Set(acceptHeader, mediaTypeFormatAndVersion)
+	resp, err := p.client.Do(req)
+	if err != nil {
+		adjustEndpointsErrorsGauge.Inc()
+		log.Debugf("Failed executing http request, %s", err)
+		return endpoints
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode != http.StatusOK {
+		adjustEndpointsErrorsGauge.Inc()
+		log.Debugf("Failed to AdjustEndpoints with code %d", resp.StatusCode)
+		return endpoints
+	}
+
+	b, err = io.ReadAll(resp.Body)
+	if err != nil {
+		adjustEndpointsErrorsGauge.Inc()
+		log.Debugf("Failed to read response body, %s", err)
+		return endpoints
+	}
+
+	err = json.Unmarshal(b, &endpoints)
+	if err != nil {
+		adjustEndpointsErrorsGauge.Inc()
+		log.Debugf("Faile to unmarshal response body, %s", err)
+		return endpoints
+	}
+	return endpoints
+}
+
+// GetDomainFilter is the default implementation of GetDomainFilter.
+func (p PluginProvider) GetDomainFilter() endpoint.DomainFilterInterface {
+	return endpoint.DomainFilter{}
```

Comment:
Why isn't this sent to the provider?

Reply:
How would you deal with returning an interface?

Reply:
Return an object implementing the interface that makes calls to the plugin.

I've simplified `DomainFilterInterface` in #3660.

Alternatively, refactor so that providers only use `DomainFilter` and have the negotiation call return the innards of `DomainFilter`.

Reply:
I lgtm'd #3660.